starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,47 @@
-function QueueDataStructure () {
+class QueueDataStructure {
+    constructor() {
+        this.queueControl = []
+        this.MAX_SIZE = 10
+
+    }
+    
+    isEmpty() {
+        if(this.queueControl.length == 0) {
+            return true
+
+        } else {
+            return false
+        }
+
+    }
+
+    canEnqueue() {
+        if(this.MAX_SIZE != this.queueControl.length) {
+            return true
+
+        } else {
+            return false
+        }
+
+    }
+
+    enqueue(parameter) {
+        if (this.MAX_SIZE == this.queueControl.length) {
+            return "Queue Overflow"
+        } else {
+            this.queueControl.unshift(parameter)
+            return this.queueControl
+
+        }
+
+    }
+    
+    dequeue() {
+        if (this.queueControl.length == 0) {
```

Comment:
La linea está bien, lo único que como arriba has hecho funciones que comparan estas cosas lo suyo es llamarlas para no hacer lo mismo todo el rato y solo llamar a la función.

la linea quedaría -->     if (this.isEmpty()) {